packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+import { Component, ComponentInterface, Prop, h, FunctionalComponent, Event, EventEmitter } from "@stencil/core";
+
+import {
+  AdvancedSelectGroupRedirect,
+  AdvancedSelectOption,
+  AdvancedSelectOptionsOrGroup,
+} from "./advanced-select.models";
+import {
+  AdvancedSelectClickEvent,
+  AdvancedSelectOptionClickEvent,
+  AdvancedSelectRedirectClickEvent,
+} from "./advanced-select.interfaces";
+
+@Component({
+  tag: "dso-advanced-select",
+  styleUrl: "advanced-select.scss",
+  shadow: true,
+})
+export class AdvancedSelect implements ComponentInterface {
+  /**
+   * The options to display in the select.
+   */
+  @Prop()
+  options: AdvancedSelectOptionsOrGroup<any>[] = [];
+
+  /**
+   * The active option. By object reference.
+   */
+  @Prop()
+  active?: AdvancedSelectOption<any>;
+
+  /**
+   * An extra text for the active option. Only visible in the list of options.
+   */
+  @Prop()
+  activeHint?: string;
+
+  /**
+   * The open state of the options list.
+   */
+  @Prop()
+  open: boolean = false;
+
+  /**
+   * Emitted when user clicks the select.
+   */
+  @Event({ bubbles: false })
+  dsoClick!: EventEmitter<AdvancedSelectClickEvent>;
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
+          <span class="active-option-label">{this.active?.label ?? "Selecteer een optie"}</span>
+          <span class="active-option-aside">
+            {this.options.some((option) => "summaryCounter" in option && option?.summaryCounter) ? (
+              <span class="badges">
+                {this.options
+                  .filter((option) => "summaryCounter" in option && option?.summaryCounter)
+                  .map((option) => {
+                    if ("options" in option) {
+                      return <dso-badge status={option.variant ?? "outline"}>{option.options.length}</dso-badge>;
+                    }
+                  })}
+              </span>
+            ) : (
+              ""
+            )}
+            <dso-icon icon="caret-down"></dso-icon>
+          </span>
+        </button>
+        <div class={{ "groups-container": true, open: this.open }}>
```

Comment:
Zie opmerking over `clsx()`.

Maar bovenal, waarom toon/verberg je dit met styling? Heb je overwogen om deze DOM tree niet te renderen?

In Angular had je waarschijnlijk `*ngIf` gebruikt.

Reply:
Ben benieuwd hoe jullie erover denken. Ik doet dit soort dingen eigenlijk zo veel mogelijk via de CSS omdat het toggle'en van een CSS-property minder performance eist dan het renderen van een DOM-tree. Vooral wanneer deze tree constant toegevoegd en verwijderd moet worden bij het openen en sluiten. Persoonlijk probeer ik ook altijd zo weinig mogelijk `*ngIf`'s te gebruiken. 

Dit doet mij wel denken aan de `will-change` CSS-property die ik nog kan toevoegen, maar zou normaal gesproken niet nodig moeten zijn.

Reply:
@tfrijsewijk Valide punt. In dat opzicht hou ik mij dan graag aan de standaard binnen DSO-Toolkit. Als dit betekend dat de DOM-tree gewoon niet gerenderd moet worden dan pas ik het aan. Als ik jou goed begrijp zou je daar voor kiezen i.p.v. verbergen met styling.

Zal ik het aanpassen?

Reply:
Graag 🙏

Reply:
Top, bij deze aangepast 👍🏻